server/client.go

```diff
@@ -1344,7 +1344,7 @@ func (c *client) closeConnection() {
 			// Forward on unsubscribes if we are not
 			// a router ourselves.
 			if c.typ != ROUTER {
-				srv.broadcastUnSubscribe(sub)
+				srv.broadcastForceUnSubscribe(sub)
```

Comment:
I think we could do without this new function by simply setting `sub.max = 0` (with a comment ;-)) above in the loop where we gather the subscriptions. We are under the client's mutex lock that protects the subscription's object. https://github.com/nats-io/gnatsd/pull/551/files#diff-853eb184ac73cf9597d7833f6b89e9c9R1320